nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,1115 @@
+
+    Attributes
+    ----------
+    volume_dimensions : array-like shape (3,)
+        "[T]he lengthss of the three volume file dimensions that are related to
+        spatial coordinates, in number of voxels. Voxel indices (which are
+        zero-based) that are used in the mapping that this element applies to
+        must be within these dimensions."
+    transformation_matrix_voxel_indices_ijk_to_xyz \
+        : Cifti2TransformationMatrixVoxelIndicesIJKtoXYZ
+        Matrix that translates voxel indices to spatial coordinates
+    """
+    def __init__(self, volume_dimensions=None, transform_matrix=None):
+        self.volume_dimensions = volume_dimensions
+        self.transformation_matrix_voxel_indices_ijk_to_xyz = transform_matrix
+
+    def _to_xml_element(self):
+        if self.volume_dimensions is None:
+            raise CIFTI2HeaderError('Volume element requires dimensions')
+
+        volume = xml.Element('Volume')
+        volume.attrib['VolumeDimensions'] = ','.join(
+            [str(val) for val in self.volume_dimensions])
+        volume.append(self.transformation_matrix_voxel_indices_ijk_to_xyz._to_xml_element())
+        return volume
+
+
+class Cifti2VertexIndices(xml.XmlSerializable, collections.MutableSequence):
+    """Cifti2 vertex indices: vertex indices for an associated brain model
+
+    The vertex indices (which are independent for each surface, and
+    zero-based) that are used in this brain model[.] The parent
+    BrainModel's ``index_count`` indicates the number of indices.
+    """
+    def __init__(self, indices=None):
+        self._indices = []
+        if indices is not None:
+            self.extend(indices)
+
+    def __len__(self):
+        return len(self._indices)
+
+    def __delitem__(self, index):
+        del self._indices[index]
+
+    def __getitem__(self, index):
+        return self._indices[index]
+
+    def __setitem__(self, index, value):
+        try:
+            value = int(value)
+            self._indices[index] = value
+        except ValueError:
+            raise ValueError('value must be an int')
+
+    def insert(self, index, value):
+        try:
+            value = int(value)
+            self._indices.insert(index, value)
+        except ValueError:
+            raise ValueError('value must be an int')
+
+    def _to_xml_element(self):
+        if len(self) == 0:
+            raise CIFTI2HeaderError('VertexIndices element requires indices')
+
+        vert_indices = xml.Element('VertexIndices')
+        vert_indices.text = ' '.join([str(i) for i in self])
+        return vert_indices
+
+
+class Cifti2BrainModel(xml.XmlSerializable):
+
+    # index_offset = int
+    # index_count = int
+    # model_type = str
+    # brain_structure = str
+    # surface_number_of_vertices = int
+    # voxel_indices_ijk = np.array
+    # vertex_indices = np.array
+
+    def __init__(self, index_offset=None, index_count=None, model_type=None,
```

Comment:
They cannot be None in the XML format. However, as I detailed above most of the correctness of the CIFTI2 XML tree is not checked on construction. And none of this was pointed out as "needed" when I decided to take on the PR.

Including that would be extremely nice but far from a negligible effort and I don't have the time to achieve it. 

I propose that, as I think that the API is stable and the parsing works, we merge it without enforcing correctness on user construction and we leave the finer checks on construction for a future one.